### PR TITLE
feat: support prettier 1/3

### LINKED ISSUE #180: template: Add support for prettier
https://prettier.io/docs/en/precommit.html#option-3-pre-commithttpsgithubcompre-commitpre-commit

additionally, in the `.editorconfig` the indentation should be set to 2:

```
indent_size = 2
```

this will impact maybe in the smoke test because it will get more issues in the files style, so it would take some time in order to to fix the issues catch by prettier.

### PR DESCRIPTION
**Pull request by Anavelyz**
## Pull Request description

Prettier is an opinionated code formatter, ensuring that all output code conforms to a consistent style.

Fixes #180

Prettier was added to the project and to the template.

**Modified files:**
- For the project:

  - .pre-commit-config.yaml
  - .prettierignore
  - .prettierrc.yaml
  - .editorconfig

- For the template:

  - src/scicookie/cookiecutter.json (and it has been reformatted)
  - src/scicookie/{{cookiecutter.project_slug}}/.pre-commit-config.yaml (and it has been reformatted)
  - src/scicookie/hooks/post_gen_project.py
  - tests/smoke/linters.sh

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [x] new feature
- [x] maintenance

About this PR:
- [x] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [x] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved

### CODE REVIEW

File: src/scicookie/{{cookiecutter.project_slug}}/docs/installation.md [hunk 1]
```diff
@@ -2,33 +2,35 @@
...
 ```
+{%- else %}
+```bash
+$ git clone #Add your repo url
+```
+{%- endif %}
 
-Or download the
-[tarball]({{ cookiecutter.git_https_upstream }}/tarball/main):
+Or download the [tarball]({{ cookiecutter.git_https_upstream }}/tarball/main):
```
Comment (Anavelyz) [inline src/scicookie/{{cookiecutter.project_slug}}/docs/installation.md:33]: @xmnlab is there a best way to do this?

File: src/scicookie/profiles/osl.yaml [hunk 2]
```diff
@@ -1,100 +1,100 @@
...
+        - ruff
+        - makim
+        - mccabe
+        - mypy
+        - pre-commit
+        - pytest
+        - hypothesis
+        - shellcheck
+        - vulture
+    visible: true
```
Comment (Anavelyz) [inline src/scicookie/profiles/osl.yaml:67]: @xmnlab Do you think we should have an alphabetical order here?

**Review by Anavelyz — COMMENTED** [review]
@xmnlab Can you have a look at it, please?

File: src/scicookie/profiles/osl.yaml (hunk 2 above)
Comment (xmnlab) [reply src/scicookie/profiles/osl.yaml:67]: I think so, alphabetical order looks good to me.

File: src/scicookie/{{cookiecutter.project_slug}}/docs-mkdocs/mkdocs.yaml [hunk 3]
```diff
@@ -1,6 +1,10 @@
 site_name: {{ cookiecutter.project_name}}
 site_url: {{ cookiecutter.project_url}}
-repo_url: {{ cookiecutter.git_https_upstream}}
+{%- if cookiecutter.git_https_upstream != "" -%}
+repo_url: {{ cookiecutter.git_https_upstream }}
+{%- else %}
+repo_url: #Add your repo url
+{%- endif %}
```
[the remark this answers is not in the document]
Comment (xmnlab) [reply src/scicookie/{{cookiecutter.project_slug}}/docs-mkdocs/mkdocs.yaml:7]: maybe a placeholder like: `https://github.com/MYORG/MYREPO`

File: src/scicookie/{{cookiecutter.project_slug}}/docs/installation.md [hunk 4]
```diff
@@ -2,33 +2,35 @@
...
 
 You can either clone the public repository:
-
+{% if cookiecutter.git_https_upstream != "" %}
 ```bash
 $ git clone {{ cookiecutter.git_https_upstream }}
 ```
+{%- else %}
+```bash
+$ git clone #Add your repo url
```
Comment (xmnlab) [inline src/scicookie/{{cookiecutter.project_slug}}/docs/installation.md:29]:
```suggestion
$ git clone https://github.com/MYORG/MYREPO
```

maybe something like that

**Review by xmnlab — CHANGES_REQUESTED** [review]
not sure what happened .. but it shouldn't change indentation from 2 spaces to 4.
maybe you will need to update  .editorconfig as well

this is what I am using for astx: https://github.com/arxlang/astx/blob/main/.editorconfig

or just maybe update the prettierc file to define the indentation space directly there.

**Comment by Anavelyz** [comment]
> not sure what happened .. but it shouldn't change indentation from 2 spaces to 4. maybe you will need to update .editorconfig as well
> 
Yes, .editorconfig has 4 spaces set: https://github.com/osl-incubator/scicookie/blob/5a6f0b44dd7d9af99063a4445a264d44bd4d2675/.editorconfig#L7

**Comment by xmnlab** [comment]
> Hi @Anavelyz!
> 
> It is usually recommended to make all the auto-formatting changes (prettier changes in this case) in a single commit. This commit's hash can then be added to `.git-blame-ignore-revs` file to ensure that the [`git blame`](https://git-scm.com/docs/git-blame) is preserved. For example, see [pybamm's .git-blame-ignore-revs](https://github.com/pybamm-team/PyBaMM/blob/develop/.git-blame-ignore-revs) file.
> 
> But this is totally up to us, if we want to follow this or not 🙂

thanks @Saransh-cpp to bring that up.

tbh I am not familiar with that .. give me a bit to check that ;)

**Comment by xmnlab** [comment]
@Saransh-cpp , I think that it makes sense. thank you again for bring that up.

so the workflow would be:

* merge this PR without any change from the prettier
* create a new PR that apply changes from prettier and merge that
* create a new PR for `.git-blame-ignore-revs`

is that right?

if so ... @Anavelyz, let's do it together ... ping me when you have a free time for that

**Comment by xmnlab** [comment]
thanks for the answer @Saransh-cpp ! appreciate that

**Review by xmnlab — APPROVED** [review]
lgtm! thanks @Anavelyz

**Pull request merged**